src/autopas/LogicHandler.h

```diff
@@ -455,12 +490,12 @@ class LogicHandler {
    * Buffer to store particles that should not yet be added to the container. There is one buffer per thread.
    * @note This buffer could potentially be replaced by a ParticleCell.
    */
-  std::vector<std::vector<Particle>> _particleBuffer;
+  std::vector<FullParticleCell<Particle>> _particleBuffer;
 
   /**
    * Buffer to store halo particles that should not yet be added to the container. There is one buffer per thread.
    * @note This buffer could potentially be replaced by a ParticleCell.
    */
-  std::vector<std::vector<Particle>> _haloParticleBuffer;
+  std::vector<FullParticleCell<Particle>> _haloParticleBuffer;
```

Comment:
Remove note on potentially removing the buffer with ParticleCell. Or do you plan to also replace the outer vector with a ParticleCell?

Also what is the advantage of using FullParticleCell here instead of vector<>?

Reply:
lol yes. This is exactly what was done in this PR -> 623ba0cc4

Reply:
The advantage is that it makes it easier to use an SoA Functor with the particles, because the cell has the logic for transforming the particle data.